src/ray/util/process.cc

```diff
@@ -63,6 +68,123 @@ int execvpe(const char *program, char *const argv[], char *const envp[]) {
 
 namespace ray {
 
+namespace {
+// Global tracker of owned child processes.
+// Only tracked in linux.
+//
+// A pid is added when a spawn is called and removed when the ProcessFD is closed (i.e.
+// it's in dtor and we no longer want to track it).
+std::mutex m;
+absl::flat_hash_set<pid_t> owned_children;
+
+void addOwnedChild(pid_t pid) {
+#ifdef __linux__
+  std::lock_guard<std::mutex> guard(m);
+  owned_children.insert(pid);
+#endif
+}
+
+void removeOwnedChild(pid_t pid) {
+#ifdef __linux__
+  std::lock_guard<std::mutex> guard(m);
+  owned_children.erase(pid);
+#endif
+}
+
+// Register a signal handler for the given signal.
+// The handler will be called with the signal_set and the error code.
+// After the handler is called, the signal will be re-registered.
+void RegisterSignalHandlerLoop(boost::asio::signal_set &signals,
+                               void (*handler)(const boost::system::error_code &, int)) {
+  signals.async_wait(
+      [&signals, handler](const boost::system::error_code &error, int signal_number) {
+        handler(error, signal_number);
+        RegisterSignalHandlerLoop(signals, handler);
```

Comment:
is it a normal practice? Is there other way to just register handler once that's called?